File: barbican/tree/branch/branch.py
from tree.status.status import Status


class Branch:

    def __init__(self):
        self.path = ''
        self.hash = ''
        self.name = ''
        self.branches = {}
        self.sources = {}
        self.headers = {}
        self.status = Status.UNCHANGED
        self.depth = 0
        self.dirty = False

File: source/tree/branch/branch.go
package branch

import (
	"github.com/inigo-selwood/barbican/tree/asset/header"
	"github.com/inigo-selwood/barbican/tree/asset/source"
)

type Branch struct {
	Name string
	Hash string

	RealRoute string
	HashRoute string

	Size int64

	Parent *Branch

	Headers map[string]*header.Header
	Sources map[string]*source.Source

	Branches map[string]*Branch
}
